src/peft/tuners/lora.py

```diff
@@ -478,26 +520,85 @@ def add_weighted_adapter(self, adapters, weights, adapter_name):
+                            current_adapter_lora_B = target.lora_B[adapter].weight
+                        elif adapter in target.lora_embedding_A:
+                            current_adapter_lora_A = target.lora_embedding_A[adapter_name]
+                            current_adapter_lora_B = target.lora_embedding_B[adapter]
+                        target_lora_A.data += current_adapter_lora_A.data * weight * target.scaling[adapter]
+                        target_lora_B.data += current_adapter_lora_B.data * weight
+                elif combination_type == "svd":
+                    delta_weight = weights[0] * target.get_delta_weight(adapter[0])
+                    for adapter, weight in zip(adapters[1:], weights[1:]):
+                        delta_weight += weight * target.get_delta_weight(adapter)
+                    conv2d = isinstance(target, Conv2d)
+                    if conv2d:
+                        conv2d_1x1 = target.weight.size()[2:4] == (1, 1)
+                        if not conv2d_1x1:
+                            delta_weight = delta_weight.flatten(start_dim=1)
+                        else:
+                            delta_weight = delta_weight.squeeze()
+                    U, S, Vh = torch.linalg.svd(delta_weight)
+                    U = U[:, :new_rank]
+                    S = S[:new_rank]
+                    U = U @ torch.diag(S)
+                    Vh = Vh[:new_rank, :]
+                    dist = torch.cat([U.flatten(), Vh.flatten()])
+                    hi_val = torch.quantile(dist, CLAMP_QUANTILE)
+                    low_val = -hi_val
+                    U = U.clamp(low_val, hi_val)
+                    Vh = Vh.clamp(low_val, hi_val)
+                    if conv2d:
+                        U = U.reshape(target_lora_B.data.shape)
+                        Vh = Vh.reshape(target_lora_A.data.shape)
+                    target_lora_A.data = Vh
+                    target_lora_B.data = U
+
+    def delete_adapter(self, adapter_name):
+        """
+        Deletes an existing adapter.
+
+        Args:
+            adapter_name (str): Name of the adapter to be deleted.
+        """
+        if adapter_name not in list(self.peft_config.keys()):
+            raise ValueError(f"Adapter {adapter_name} does not exist")
+        del self.peft_config[adapter_name]
+        key_list = [key for key, _ in self.model.named_modules() if "lora" not in key]
+        for key in key_list:
+            _, target, _ = _get_submodules(self.model, key)
+            if isinstance(target, LoraLayer):
+                for attr in [
+                    "r",
+                    "lora_alpha",
+                    "scaling",
+                    "lora_A",
+                    "lora_B",
+                    "lora_embedding_A",
+                    "lora_embedding_B",
+                    "lora_dropout",
+                ]:
+                    if adapter_name in getattr(target, attr):
```

Comment:
Just wondering: Could a valid adapter name be a substring of another valid adapter name? If yes, could this `in` check result in false positives?

Reply:
Hello, here we are checking in element is present in a list rather than if a string is a substring of another string. So, this should only match if the adapter_name is exactly present in the dict keys.

Reply:
I see, makes sense.